src/display/editor/outliner.js

```diff
@@ -0,0 +1,251 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class Outliner {
+  #box;
+
+  #verticalEdges = [];
+
+  #intervals = [];
+
+  constructor(boxes, extra = 0, isLTR = true) {
```

Comment:
What is `extra`, since that's not immediately clear when reading this line?
Please either use a slightly more verbose variable name, or at least add a JSDoc to explain things here.